modules/tinymce/src/core/main/ts/fmt/FormatUtils.ts

```diff
@@ -18,39 +18,73 @@ const isNode = (node: any): node is Node => !!(node).nodeType;
 const isElementNode = (node: Node): node is Element =>
   NodeType.isElement(node) && !Bookmarks.isBookmarkNode(node) && !isCaretNode(node) && !NodeType.isBogus(node);
 
-const isInlineBlock = (node: Node): boolean => {
-  return node && /^(IMG)$/.test(node.nodeName);
+// Control blocks are any elements e.g. (img, iframe, noneditable element) that can be selected individually
+// In TinyMCE, selected blocks are indicated with the data-mce-selected attribute
+const isControlBlockSelected = (dom: DOMUtils, node: Node): boolean => {
+  // Table cells are a special case and are separately handled from native editor selection
+  if (isElementNode(node) && !/^(TD|TH)$/.test(node.nodeName)) {
+    const selectedAttr = dom.getAttrib(node, 'data-mce-selected');
+    const value = parseInt(selectedAttr, 10);
+    // Avoid cases where data-mce-selected is not a positive number e.g. inline-boundary
+    return !isNaN(value) && value > 0;
+  } else {
+    return false;
+  }
 };
 
 const isEditable = (elm: HTMLElement): boolean =>
   elm.isContentEditable === true;
 
-const moveStart = (dom: DOMUtils, selection: EditorSelection, rng: Range): void => {
-  const offset = rng.startOffset;
-  let container = rng.startContainer;
+const adjustSelectionAfter = <T>(editor: Editor, action: () => T, shouldMoveStart: (startNode: Node) => boolean): void => {
+  const { selection, dom } = editor;
+  const selectedNodeBeforeAction = selection.getNode();
+  const isSelectedBeforeNodeNoneditable = dom.getContentEditable(selectedNodeBeforeAction) === 'false';
```

Comment:
Probably a bit shorter when minifed.

```suggestion
  const isSelectedBeforeNodeNoneditable = NodeType.isContentEditableFalse(selectedNodeBeforeAction);
```